### PR TITLE
Updating the UI to reflect the access policies being enforced

### PR DESCRIPTION
**Pull request by mcgilman**
NIFI-2307:
- Enforcing connection permissions based on the source and destination component.
- Removing connection specific access policies.

NIFI-2265:
- Filtering out sensitive details in component status and status history when appropriate.

NIFI-1800:
- Adding parent process group id to the Controller Services table.

NIFI-2077:
- Removing some old un-used icons following the UI refresh.

NIFI-2242:
- Requiring write permissions for all components in a selection.

NIFI-2080:
- Updating style of the name in the selection context to handle scroll bars and use available width.

NIFI-2331:
- Addressing issue when removing a user/group which was causing the tenant policy to be removed.

NIFI-2335:
- Ensuring the flow is saved after starting/stopping a process group.

NIFI-2235:
- Ensuring we use consistent conditions between the context menu and the operate palette.

- Allowing users with read only access to the tenants page.
- Fixing current user integration test.
- Ensuring schedule methods are locked appropriately.

### CODE REVIEW

**Comment by JPercivall** [comment]
Reviewing

**Comment by JPercivall** [comment]
Doesn't seem to address NIFI-2242. I have write permission to GenerateFF but not LogAttribute. This UI mess up fixed itself after refresh.

screen shot 2016-07-21 at 12 38 31 pm

**Comment by mcgilman** [comment]
NIFI-2242 was when there are multiple components selected with mixed permissions so I think this may be a slightly different case but I'd be happy to address as part of this PR.

**Comment by JPercivall** [comment]
I don't see how this is different, maybe I mis-explained it the first time:

LogAttribute I don't have write access
GenerateFF I do
I select all three (including conneciton) and attempt to move together
Get pop-up saying i can't modify LogAttribute
GenerateFF moves and UI looks like that

**Comment by JPercivall** [comment]
Actually I think that was another bug relating to inconsistent cluster state. I just retried and got a pop-up saying "Must be authorized to modify every component selected."

**Comment by mcgilman** [comment]
Yeah, I'm unable to replicate. The first message you mentioned sounds like the request was attempted like it was prior to the fix. Maybe your browser had some old JS cached. The "Must be authorized..." message is a client side check that I added to address NIFI-2242. Let me know if you still seeing any weirdness.

**Comment by JPercivall** [comment]
One thing I did notice is when I have selected just GenerateFF I can move it (UI properly moves relationship too) but if I have GenerateFF and connection selected it fails (message "Must be authorized to modify every component selected."). I believe this is because it is checking if I have permission to move the connection which in turn checks if I have permission to modify the destination (which I don't).

**Comment by mcgilman** [comment]
That is correct. Permissions of the connection are driven by the permissions of the source and destination.

**Comment by JPercivall** [comment]
I understand why, just seems odd though because the action "moving GenerateFF" vs "having GenerateFF and connection selected, attempt to move them" has the same end result.

**Comment by mcgilman** [comment]
It's not the same. Moving a connection would move it's bend points. Moving the source does not affect the connection. The start and end points are dynamic and based on the position of the source and destination.

**Comment by JPercivall** [comment]
Created a new ticket for an issue I saw:
https://issues.apache.org/jira/browse/NIFI-2347

**Comment by JPercivall** [comment]
Relating to NIFI-2265. I have a process group that I do not have view permissions on. I have processors in that group that inherit the policy from the group. When I view the flow on the canvas or by clicking "stats" I properly don't the name but when I view "summary" (from the top menu) I see the names. 

Tracking it down in the developer tools, it looks like it comes from the ProcessGroupStatusSnapshots (I don't have view access to that group either).

**Comment by mcgilman** [comment]
Good catch. Will update PR shortly. Thanks!

**Comment by JPercivall** [comment]
I don't think NIFI-2235 was addressed. I am still able to reproduce

**Comment by JPercivall** [comment]
NIFI-2235 offending logic path:
copy processor in root PG
go into PG I do not have modify access
Context menu does not have paste, palette menu allows it to be clicked
after i click paste (and get no access) it correctly refreshes to show no ability to paste

**Comment by JPercivall** [comment]
Similar logic is possible when going from the sub group to root group.

**Comment by JPercivall** [comment]
As part of NIFI-1800, when viewing the config for a process group it should have the UUID at the top. This way at the very least I can tell which controller services belong to this ProcessGroup without trying to remember the exact UUID (no tab in that panel shows that information).

Also it may be best to have the name in that column instead of UUID.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java [hunk 1]
```diff
@@ -2147,9 +2147,15 @@ public ProcessGroupStatus getGroupStatus(final ProcessGroup group, final Reposit
             return null;
         }
 
+        // if the user is null, this is not a web request so we need to treat as authorized - this is the case when
+        // gather status for history in a background thread... we then will check the access of the user when pulling
+        // back history...
+        final NiFiUser user = NiFiUserUtils.getNiFiUser();
+        final boolean isAuthorized = user == null ? true : group.isAuthorized(authorizer, RequestAction.READ, user);
```
Comment (JPercivall) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java]: This seems like a very insecure way to do things (if null then authorized).

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/authorization/ConnectionAuthorizable.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * limitations under the License.
+ */
+package org.apache.nifi.authorization;
+
+import org.apache.nifi.authorization.resource.Authorizable;
+import org.apache.nifi.connectable.Connectable;
+import org.apache.nifi.groups.ProcessGroup;
+
+/**
+ * Authorizable for policies of an Authorizable.
```
Comment (JPercivall) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/authorization/ConnectionAuthorizable.java]: I think this should be "Authorizable for policies of a Connection."

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java (hunk 1 above)
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java]: When handling a web request, the user will always be available. When this method is running outside the scope of a web request, there is nothing to attempt to authorize (nor should we). This conditional authorization seems appropriate to me. However, I think we can clean up the code to make the distinction a little more clear and deliberate.

**Comment by JPercivall** [comment]
+1

Visually verified code and comments were addressed. Did a contrib check build and on a 3 node secure cluster ran through each ticket and did many different iterations of various paths of logic. Thanks Matt @mcgilman, I will merge it in.

I will be creating a ticket for the controller service/process group Id because there is another problem with that panel (without W permissions the process group details don't show up).

**Comment by JPercivall** [comment]
Created NIFI-2358 for the issues I found with PG config panel.

https://issues.apache.org/jira/browse/NIFI-2358

**Comment by JPercivall** [comment]
Github is taking a long time to replicate but here is a link to the commit in apache (I pushed it, I swear, lol):

https://git-wip-us.apache.org/repos/asf?p=nifi.git;a=commit;h=4a4d60e6af5014342039f751c7b058079d643f7b

**Pull request closed without merging**